source/slang/slang-ir.cpp

```diff
@@ -2809,6 +2809,13 @@ namespace Slang
             operands);
     }
 
+    IRVectorType* IRBuilder::getVectorType(
+        IRType* elementType,
+        UInt    elementCount)
+    {
+        return getVectorType(elementType, getIntValue(getUIntType(), elementCount));
```

Comment:
I think in everywhere else we are using getIntType for the index operand.
We should keep things consistent by emiting an IntType here to allow redundancy removal.

Reply:
Maybe use `IRIntegerType` for the `elementCount`  parameter.

Reply:
good idea, thanks